src/ProjectConfiguration.ts

```diff
@@ -617,6 +624,50 @@ export function validateCompanionDefaultWakeInterval(
   return diagnostics;
 }
 
+export function validateTileBuildTarget( tile: Tile, { buildTargets }: AppProjectConfiguration) {
+  if (tile.buildTargets !== undefined){
+    const diagnostic = constrainedSetDiagnostics({
+      actualValues: tile.buildTargets,
+      knownValues: buildTargets,
+      valueTypeNoun: 'tile build targets',
+      notFoundIsFatal: true
+    })
+    return diagnostic;
+  }
+
+  // Return empty list. Nothing to check
+  return new DiagnosticList();
+}
+
+export function validateTileUUID(tile: Tile, config: AppProjectConfiguration) {
+  const diagnostic = new DiagnosticList()
+  if (tile.uuid === undefined) {
+    diagnostic.pushFatalError("Tile UUID can't be undefined");
+  }
+
+  if (!validator.isUUID(String(tile.uuid))) {
+    diagnostic.pushFatalError(
+      'Tile UUID must be a valid UUID',
+    );
+  }
+
+  const knownUUID = [config.appUUID]
+  config.tiles!.forEach((tile) => knownUUID.push(tile.uuid))
+  if (knownUUID.indexOf(tile.uuid) !== knownUUID.lastIndexOf(tile.uuid)) {
+    diagnostic.pushFatalError("Can't have duplicate UUIDs between app and tiles");
```

Comment:
This shouldn't necessarily be true. My recommendation has been for tiles to have their own ID namespace, which would be at odds with this check. I don't think we should be making that part of the toolchain if some users decide to make this their own convention.

Reply:
Agreed. Parent uuid + tile uuid should be unique though. I'll make the changes.

Reply:
Made the change that just tile uuids are unique.